linera-service/tests/end_to_end_tests.rs

```diff
@@ -1634,6 +1661,247 @@ async fn test_end_to_end_crowd_funding() {
     node_service2.assert_is_running();
 }
 
+#[cfg(any(feature = "wasmer", feature = "wasmtime"))]
+#[test_log::test(tokio::test)]
+async fn test_end_to_end_matching_engine() {
+    use fungible::{FungibleTokenAbi, InitialState};
+    use matching_engine::{OrderNature, Parameters, Price};
+
+    let _guard = INTEGRATION_TEST_GUARD.lock().await;
+
+    let network = Network::Grpc;
+    let mut runner = TestRunner::new(network, 4);
+    let client1 = runner.make_client(network);
+    let client2 = runner.make_client(network);
+    let client3 = runner.make_client(network);
+
+    runner.generate_initial_validator_config().await;
+    client1.create_genesis_config().await;
+    client2.wallet_init(&[]).await;
+    client3.wallet_init(&[]).await;
+
+    // Create initial server and client config.
+    runner.run_local_net().await;
+    let (contract_fungible, service_fungible) = runner.build_example("fungible").await;
+    let (contract_matching, service_matching) = runner.build_example("matching_engine").await;
+
+    let chain1 = client1.get_wallet().default_chain().unwrap();
+    let chain2 = client1.open_and_assign(&client2).await;
+    let chain3 = client1.open_and_assign(&client3).await;
+
+    // The players
+    let owner_matching = client1.get_account_owner();
+    let owner_a = client2.get_account_owner();
+    let owner_b = client3.get_account_owner();
```

Comment:
Should we call the clients `client_matching`, `client_a` and `client_b` then?

Reply:
I was thinking about that and I preferred `client_admin`, `client_a`, `client_b`.

Reply:
Not sure about using "admin" for chains other than the system's admin chain (that determines the validators and stuff).